### DISCUSSION TITLE
TIMEOUT error while installing moose conda packages

### DISCUSSION THREAD

**Discussion by amintgn** [idaholab/moose#23402] [Q&A Getting Started] [answered]
Dear all,

I'm a beginner in moose and c++. While installing moose conda packages using the command "" I receive the following error.
I appreciate your help. Here is what I receive:

```pre
Looking for: ['moose-tools', 'moose-libmesh']

[+] 9.0s
https://conda.software.inl.gov/public/linux-64      ??.?MB @  ??.?MB/s 0 failed  9.0s
Download error (28) Timeout was reached [https://conda.software.inl.gov/public/noarch/repodata.json]
Connection timeout after 9000 ms

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/conda/exceptions.py", line 1118, in __call__
        return func(*args, **kwargs)
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 936, in exception_converter
        raise e
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 929, in exception_converter
        exit_code = _wrapped_main(*args, **kwargs)
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 887, in _wrapped_main
        result = do_call(parsed_args, p)
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 754, in do_call
        exit_code = create(args, parser)
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 630, in create
        return install(args, parser, "create")
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 497, in install
        index = load_channels(pool, channels, repos)
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/utils.py", line 129, in load_channels
        index = get_index(
      File "/home/amin/mambaforge3/lib/python3.10/site-packages/mamba/utils.py", line 110, in get_index
        is_downloaded = dlist.download(api.MAMBA_DOWNLOAD_FAILFAST)
    RuntimeError: Download error (28) Timeout was reached [https://conda.software.inl.gov/public/noarch/repodata.json]
    Connection timeout after 9000 ms

`$ /home/amin/mambaforge3/bin/mamba create -n moose moose-tools moose-libmesh`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/home/amin/mambaforge3/bin/conda
             CONDA_PREFIX=/home/amin/mambaforge3
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/home/amin/mambaforge3/bin/python
               CONDA_ROOT=/home/amin/mambaforge3
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>
               LD_PRELOAD=<not set>
                     PATH=/home/amin/mambaforge3/bin:/home/amin/mambaforge3/condabin:/usr/local/
                          sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/loca
                          l/games:/usr/lib/wsl/lib:/mnt/c/Program Files/WindowsApps/MicrosoftCor
                          porationII.WindowsSubsystemForLinux_1.0.3.0_x64__8wekyb3d8bbwe:/mnt/c/
                          Program Files/Microsoft MPI/Bin:/mnt/c/Program Files/Microsoft HPC
                          Pack 2012/Bin:/mnt/c/Program Files
                          (x86)/CMG/CMGJobService:/mnt/c/Program Files (x86)/Intel/MPI-
                          RT/5.0.2.044/intel64/bin:/mnt/c/Program Files (x86)/Intel/iCLS
                          Client:/mnt/c/Program Files/Intel/iCLS Client:/mnt/c/WINDOWS/system32:
                          /mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/Wi
                          ndowsPowerShell/v1.0:/mnt/c/Program
                          Files/MATLAB/R2018a/runtime/win64:/mnt/c/Program Files/MATLAB/R2018a/b
                          in:/mnt/c/ecl/HOME:/mnt/c/ecl/MACROS:/mnt/c/WINDOWS/System32/OpenSSH:/
                          mnt/c/Program Files (x86)/Schlumberger/OLGA-S
                          2017.2.0/Win32:/mnt/c/Program Files (x86)/Schlumberger/OLGA-S
                          2017.2.0/Win64:/mnt/c/Program Files
                          (x86)/Hummingbird/Connectivity/11.00/Accessories:/mnt/c/Program Files
                          (x86)/Intel/Intel(R) Management Engine Components/DAL:/mnt/c/Program
                          Files/Intel/Intel(R) Management Engine Components/DAL:/mnt/c/Program
                          Files (x86)/Intel/Intel(R) Management Engine
                          Components/IPT:/mnt/c/Program Files/Intel/Intel(R) Management Engine
                          Components/IPT:/mnt/c/TDM-GCC-
                          64/bin:/mnt/c/Users/Amin/AppData/Local/Microsoft/WindowsApps:/mnt/c/Us
                          ers/Amin/AppData/Local/Programs/Microsoft VS Code/bin:/snap/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : base
    active env location : /home/amin/mambaforge3
            shell level : 1
       user config file : /home/amin/.condarc
 populated config files : /home/amin/mambaforge3/.condarc
                          /home/amin/.condarc
          conda version : 22.11.1
    conda-build version : not installed
         python version : 3.10.8.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.31=0
                          __linux=5.15.79.1=0
                          __unix=0=0
       base environment : /home/amin/mambaforge3  (writable)
      conda av data dir : /home/amin/mambaforge3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.software.inl.gov/public/linux-64
                          https://conda.software.inl.gov/public/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/amin/mambaforge3/pkgs
                          /home/amin/.conda/pkgs
       envs directories : /home/amin/mambaforge3/envs
                          /home/amin/.conda/envs
               platform : linux-64
             user-agent : conda/22.11.1 requests/2.28.2 CPython/3.10.8 Linux/5.15.79.1-microsoft-standard-WSL2 ubuntu/20.04.3 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.
```

**Comment by GiudGiud** [comment]
Hello

There s unfortunately no way I know to recover from borked conda environments.

you ll have to remove it and try again. And if it doesn’t work then remove mamba and follow our instructions for getting started from the beginning again

Guillaume

**Reply by amintgn** [reply]
Thanks Guillaume.
Unfortunately, I had no progress in installation. I am a Windows user and working with WSL is really new to me. I cannot even remove mamba which is a little confusing for me and I have no clue how to do so. I tried "conda remove mamba" but it is stuck and shows only: 
Collecting package metadata (repodata.json):

failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://conda.software.inl.gov/public/linux-64/repodata.json>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
'https//conda.software.inl.gov/public/linux-64'



I appreciate your help if you can provide me with information on how to progress.

Amin

**Reply by GiudGiud** [reply, marked as answer]
Hello

Just delete the folder where mamba / conda is installed.
It should be mamba if you followed the online instructions

Guillaume

**Reply by milljm** [reply]
```bash
rm -rf /home/amin/mambaforge3
```
Should delete Conda... Once you do the above, close out of WSL, and re-open WSL. Then try to run through the installation steps again.

Note: The second time you received an error connecting to https://conda.software.inl.gov/public/linux-64/repodata.json, could actually have been _my fault_.
At 8:36 AM MST, I deleted much of the indexing cache that was present to build anew. It's possible you attempted to connect during that time frame.

**Comment by amintgn** [comment]
Hi

I removed and re-installed mamba successfully, but the first problem (TIMEOUT error) is still the case.
I reckon it's related to my internet connection. I found the following guide on MOOSE website (Troubleshooting guide for conda):

*************************************************************************************************************************************************
TIMEOUT

If your tests fail due to timeout errors, it is most likely you have a good installation, but a slow machine (or slow filesystem). You can adjust the amount of time that the TestHarness allows a test to run before timing out, by adding a paramater to your test file:

[Tests]
  [./timeout]
    type = RunApp
    input = my_input_file.i
    max_time = 300   <-- time in seconds before a timeout occurs. 300 is the default for all tests.
  [../]
[]
*************************************************************************************************************************************************

But I have no clue how should I implement such a code.

Amin

**Reply by milljm** [reply]
The troubleshooting steps you are referring to are for `./run_tests` test timeouts. Are you saying you are still receiving something similar to:
```pre
https://conda.software.inl.gov/public/linux-64      ??.?MB @  ??.?MB/s 0 failed  9.0s
Download error (28) Timeout was reached [https://conda.software.inl.gov/public/noarch/repodata.json]
Connection timeout after 9000 ms
```

**Reply by amintgn** [reply]
Yes. This is what I receive after running "mamba create -n moose moose-tools moose-libmesh" after around 9 seconds. I'm really wondering what could cause this issue.

**Reply by milljm** [reply]
What happens when you:

```bash
curl https://conda.software.inl.gov/public
```

This is a simple bash command that reads a url, and prints the contents on the screen. A successful run will look like:

```pre
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>301 Moved Permanently</title>
</head><body>
<h1>Moved Permanently</h1>
<p>The document has moved <a href="http://conda.software.inl.gov/public/">here</a>.</p>
</body></html>
```

**Reply by amintgn** [reply]
It returns:
`curl: (28) Failed to connect to conda.software.inl.gov port 443: Connection timed out`

**Reply by milljm** [reply]
Are you able to browse to the following address using your browser on the same machine: https://conda.software.inl.gov/public/

**Reply by amintgn** [reply]
Yes, I am. Out of the WSL everything works and I can access the link above.

**Reply by milljm** [reply]
I am not sure why your WSL would not allow you to that address, while the host machine does. Is this some work machine at a lab or some other location which may lay down access restriction rules or some such??

**Reply by amintgn** [reply]
I should say it's my personal machine. But the thing is there are some restrictions in my location (Iran) to access some websites. I think something is wrong with my internet connection and I may try this some time later. Internet quality and restrictions fluctuate in my location from time to time.

**Reply by milljm** [reply]
I am betting your Windows machine is routing through a Proxy, while the WSL guest machine is not configured to do so. I don't know how you would figure out your Proxy settings in Windows, but if you can figure them out; you would want to export those addresses like so:

```bash
export http_proxy=<your proxy server>
export https_proxy=<your proxy server>
```
Ref: https://stackoverflow.com/questions/9445489/performing-http-requests-with-curl-using-proxy

Conda will use the same environment variables... Edit: That is to say, once you can get:

```bash
export http_proxy=<your proxy server>
export https_proxy=<your proxy server>
curl https://conda.software.inl.gov/public
```
to work, so should `mamba` commands.

**Reply by amintgn** [reply]
Yes, I have to use a Proxy. I tried several ways to to use curl with a proxy bud all of them fail to connect to my Proxy server. It really beats me.
